### ISSUE TITLE
torch.nn.InstanceNorm1d has inconsistent semantics of train and eval

### ISSUE THREAD

**Issue by The-Fonz** [pytorch/pytorch#4348]
For `torch.nn.InstanceNorm1d` the docs say:

*At evaluation time (.eval()), the default behaviour of the InstanceNorm module stays the same i.e. running mean/variance is NOT used for normalization. One can force using stored mean and variance with .train(False) method.*

However, the source for `.eval()` shows that it simply calls `.train(False)`.

**Comment by apaszke** [comment]
It's because of [these lines](https://github.com/pytorch/pytorch/blob/master/torch/nn/modules/instancenorm.py#L35-L36). However, I'd say this is a questionable design decision and we should definitely change this. `train(False)` should always be sematically equivalent to `eval()`!

**Comment by SsnL** [comment]
@apaszke I agree. We should fix `train(False)` to be equivalent with `eval()`. But should we also provide an additional method on `InstanceNorm*d` for users who want to use running mean/var?

**Comment by apaszke** [comment]
Sure, an extra method sounds good.
